### ISSUE TITLE
[BUG]: Terminal emulator produces erroneous output with unicode

### ISSUE THREAD

**Issue by mattcalabrese** [compiler-explorer/compiler-explorer#8085]
### Describe the bug

When streaming out unicode characters to stdout there is occasionally erroneous output when run in compiler explorer. It happens with different compilers (tested gcc and clang, libstdc++ and libc++) and with iostreams, printf, std::print. I have not noticed this before so it may be new, but I also rarely print unicode so maybe it was just latent. The issue is nondeterministic, so you may need multiple runs in order to replicate, but it happens fairly consistently.

### Steps to reproduce

A std::print of many ┼ characters should trigger it eventually, but I have seen it with when printing other unicode characters, such as ♣♦♥♠. Reproduction link has a simple example.

### Expected behavior

Consistent printing of the correct unicode characters in the terminal.

### Reproduction link

https://godbolt.org/z/7ed1zaGs4

### Screenshots

<img width="1133" height="965" alt="Image" src="https://github.com/user-attachments/assets/5ede4d2e-4a29-4d18-8379-83b0ea0d05e7" />

### Browser version

Firefox 142.0.1 (64-bit)

**Comment by partouf** [comment]
Seems to not be happening when using up to 3 iterations. Guessing it has something to do with a buffer.
https://godbolt.org/z/M5bMzK5T3

If you add a flush in between each iteration it's fine no matter how much iterations there are because everything is flushed before it can break.

I don't think this is a CE thing, but just the stream not knowing which characters belong to each other.

**Comment by Sir-Walrus** [comment]
Issue reproduces with
```c
#include <unistd.h>

int main()
{
	for (int i=0;i<5;i++)
	{
		write(1, "\xC3", 1);
		usleep(10000);
		write(1, "\xB8", 1);
		usleep(10000);
		write(1, "\xC3\xB8", 2);
		usleep(10000);
	}
}
```
https://godbolt.org/z/Wjj7Mqc6E

On mate-terminal. this prints øøøøøøøøøø.
On Godbolt, it's ��ø��ø��ø��ø��ø.

Whoever's decoding the process' stdout from byte stream to utf8 seems to be a bit overeager; stdout should be a byte stream, it shouldn't care about message boundaries.

**Comment by partouf** [comment]
if I had to point at a culprit, it's probably our stream appending thing https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/exec.ts#L166

**Comment by partouf** [comment]
https://stackoverflow.com/a/49428486

**Comment by Sir-Walrus** [comment]
I can't find a clear answer in the docs for what encoding child_process.spawn() uses. .exec() [defaults to](https://nodejs.org/api/child_process.html#child_processexeccommand-options-callback) encoding:'utf8' (and 'buffer' would be immune to this problem), but that option is not documented for spawn().

Sounds to me like high probability of some kind of bug in node:child_process, but I can't prove anything; I don't have Node available on my machine.

**Comment by partouf** [comment]
This is now live
